File: js/schoolNames.js
var names1 = ["Acadia","Angelwood","Apple Valley","Bayshore","Bear Mountain","Bear River","Bear Valley","Big Pine","Big Valley","Blue River","Broad River","Canyon View","Cape Coral","Central","Central Valley","Clear Lake","Clearwater","Columbus","Copper Cove","Coral Coast","Coral Springs","Crossroads","Crystal River","Cypress","Da Vinci","Darwin","Deer River","Deer Valley","Desert Sands","Desert Winds","Diamond","Eagle Mountain","East Bridge","East Shores","Eastside","Eastview","Eastwood","Edgewater","Edgewood","Einstein","Elk Creek","Elk Grove","Elk Valley","Enterprise","Eureka","Evergreen","Faith","Faraday","Foothill","Forest Lake","Fortuna","Freedom","Frozen Lake","Garden Grove","Golden Oak","Golden Sierra","Golden Valley","Grand Mountain","Grand Ridge","Grandview","Granite Bay","Granite Hills","Grapevine","Great Oak","Green Meadows","Green Valley","Greenfield","Greenlands","Greenville","Harbor View","Hawking","Hercules","Heritage","Highland","Hillview","Holy Oaks","Horizon","Independence","Laguna Bay","Laguna Beach","Laguna Creek","Lakeside","Lakewood","Liberty","Lincoln","Little Valley","Littlerock","Littlewood","Lone Oak","Lone Pine","Long Beach","Mammoth","Maple Hills","Maple Leaf","Maple Park","Maple Ridge","Marble Hills","Marie Curie","Martin Luther King","Meadows","Meadows Ridge","Millenium","Monarch","Mountain Oak","Mountainridge","Mountainview","Northride","Northview","Oak Grove","Oak Hills","Oak Park","Oak Ridge","Oakland","Oakleaf","Oakwood","Ocean View","Oceanside","Oyster Harbour","Pacific Grove","Palm Valley","Panorama","Paradise","Paramount","Patriot","Pine Hill","Pine Hills","Pinewood","Pioneer","Pleasant Grove","Pleasant Hill","Pleasant Valley","Promise","Providence","Rainbow","Ravenwood","Redlands","Redwood","Ridgeview","River Fork","River Valley","Riverbank","Riverdale","Riverview","Rutherford","Sacred Heart","Saint Helena","Saint Mary's","Sandalwood","Savanna","Seacoast","Seal Bay","Seal Coast","Seal Gulch","Seaside","Sierra","Silver Creek","Silver Oak","Silver Valley","Silverleaf","Skyline","Somerset","South Fork","Southview","Spring Gardens","Spring Hill","Springfield","Stonewall","Storm Coast","Summerfield","Summers","Summerville","Summit","Sun Valley","Sunny Coast","Sunny Hills","Sunnyside","Sunset","Sunshine","Timber Creek","Tranquillity","Trinity","Upper Lake","Valley View","Village","Vista","Waterfall","Waterfalls","Waterford","West Bridge","West Shores","Westside","Westview","Westwood","Whale Coast","Whale Gulch","White Mountain","Whitewater","Wildwood","Willow","Willow Creek","Willow Park","Winters","Winterville","Woodcreek","Woodside"];
var names2 = ["Academy","College","High","High School","School","University","Elementary","Middle School","Institute","Charter School","School of Fine Arts","Secondary School","School for Girls","School for Boys","Conservatory","Grammar School","Kindergarten","Technical School"];
var br = "";

function nameGen(){
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.floor(Math.random() * names1.length);
		rnd2 = Math.floor(Math.random() * names2.length);
		names = names1[rnd] + " " + names2[rnd2];
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/schoolNames.py
__all__ = ['schoolNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['names1', 'names2', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('names1').get('length'))))
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('names2').get('length'))))
            var.put('names', ((var.get('names1').get(var.get('rnd'))+Js(' '))+var.get('names2').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('names1', Js([Js('Acadia'), Js('Angelwood'), Js('Apple Valley'), Js('Bayshore'), Js('Bear Mountain'), Js('Bear River'), Js('Bear Valley'), Js('Big Pine'), Js('Big Valley'), Js('Blue River'), Js('Broad River'), Js('Canyon View'), Js('Cape Coral'), Js('Central'), Js('Central Valley'), Js('Clear Lake'), Js('Clearwater'), Js('Columbus'), Js('Copper Cove'), Js('Coral Coast'), Js('Coral Springs'), Js('Crossroads'), Js('Crystal River'), Js('Cypress'), Js('Da Vinci'), Js('Darwin'), Js('Deer River'), Js('Deer Valley'), Js('Desert Sands'), Js('Desert Winds'), Js('Diamond'), Js('Eagle Mountain'), Js('East Bridge'), Js('East Shores'), Js('Eastside'), Js('Eastview'), Js('Eastwood'), Js('Edgewater'), Js('Edgewood'), Js('Einstein'), Js('Elk Creek'), Js('Elk Grove'), Js('Elk Valley'), Js('Enterprise'), Js('Eureka'), Js('Evergreen'), Js('Faith'), Js('Faraday'), Js('Foothill'), Js('Forest Lake'), Js('Fortuna'), Js('Freedom'), Js('Frozen Lake'), Js('Garden Grove'), Js('Golden Oak'), Js('Golden Sierra'), Js('Golden Valley'), Js('Grand Mountain'), Js('Grand Ridge'), Js('Grandview'), Js('Granite Bay'), Js('Granite Hills'), Js('Grapevine'), Js('Great Oak'), Js('Green Meadows'), Js('Green Valley'), Js('Greenfield'), Js('Greenlands'), Js('Greenville'), Js('Harbor View'), Js('Hawking'), Js('Hercules'), Js('Heritage'), Js('Highland'), Js('Hillview'), Js('Holy Oaks'), Js('Horizon'), Js('Independence'), Js('Laguna Bay'), Js('Laguna Beach'), Js('Laguna Creek'), Js('Lakeside'), Js('Lakewood'), Js('Liberty'), Js('Lincoln'), Js('Little Valley'), Js('Littlerock'), Js('Littlewood'), Js('Lone Oak'), Js('Lone Pine'), Js('Long Beach'), Js('Mammoth'), Js('Maple Hills'), Js('Maple Leaf'), Js('Maple Park'), Js('Maple Ridge'), Js('Marble Hills'), Js('Marie Curie'), Js('Martin Luther King'), Js('Meadows'), Js('Meadows Ridge'), Js('Millenium'), Js('Monarch'), Js('Mountain Oak'), Js('Mountainridge'), Js('Mountainview'), Js('Northride'), Js('Northview'), Js('Oak Grove'), Js('Oak Hills'), Js('Oak Park'), Js('Oak Ridge'), Js('Oakland'), Js('Oakleaf'), Js('Oakwood'), Js('Ocean View'), Js('Oceanside'), Js('Oyster Harbour'), Js('Pacific Grove'), Js('Palm Valley'), Js('Panorama'), Js('Paradise'), Js('Paramount'), Js('Patriot'), Js('Pine Hill'), Js('Pine Hills'), Js('Pinewood'), Js('Pioneer'), Js('Pleasant Grove'), Js('Pleasant Hill'), Js('Pleasant Valley'), Js('Promise'), Js('Providence'), Js('Rainbow'), Js('Ravenwood'), Js('Redlands'), Js('Redwood'), Js('Ridgeview'), Js('River Fork'), Js('River Valley'), Js('Riverbank'), Js('Riverdale'), Js('Riverview'), Js('Rutherford'), Js('Sacred Heart'), Js('Saint Helena'), Js("Saint Mary's"), Js('Sandalwood'), Js('Savanna'), Js('Seacoast'), Js('Seal Bay'), Js('Seal Coast'), Js('Seal Gulch'), Js('Seaside'), Js('Sierra'), Js('Silver Creek'), Js('Silver Oak'), Js('Silver Valley'), Js('Silverleaf'), Js('Skyline'), Js('Somerset'), Js('South Fork'), Js('Southview'), Js('Spring Gardens'), Js('Spring Hill'), Js('Springfield'), Js('Stonewall'), Js('Storm Coast'), Js('Summerfield'), Js('Summers'), Js('Summerville'), Js('Summit'), Js('Sun Valley'), Js('Sunny Coast'), Js('Sunny Hills'), Js('Sunnyside'), Js('Sunset'), Js('Sunshine'), Js('Timber Creek'), Js('Tranquillity'), Js('Trinity'), Js('Upper Lake'), Js('Valley View'), Js('Village'), Js('Vista'), Js('Waterfall'), Js('Waterfalls'), Js('Waterford'), Js('West Bridge'), Js('West Shores'), Js('Westside'), Js('Westview'), Js('Westwood'), Js('Whale Coast'), Js('Whale Gulch'), Js('White Mountain'), Js('Whitewater'), Js('Wildwood'), Js('Willow'), Js('Willow Creek'), Js('Willow Park'), Js('Winters'), Js('Winterville'), Js('Woodcreek'), Js('Woodside')]))
var.put('names2', Js([Js('Academy'), Js('College'), Js('High'), Js('High School'), Js('School'), Js('University'), Js('Elementary'), Js('Middle School'), Js('Institute'), Js('Charter School'), Js('School of Fine Arts'), Js('Secondary School'), Js('School for Girls'), Js('School for Boys'), Js('Conservatory'), Js('Grammar School'), Js('Kindergarten'), Js('Technical School')]))
pass
pass


# Add lib to the module scope
schoolNames = var.to_python()